### ISSUE TITLE
404 when updating on windows due to flash-attention

### ISSUE THREAD

**Issue by CamiloMM** [oobabooga/text-generation-webui#4369]
### Describe the bug

I get a 404 when updating, which fails the update:

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Running `update_windows.bat`

### Logs

```shell
Ignoring flash-attn: markers 'platform_system == "Windows" and python_version == "3.11"' don't match your environment
Collecting flash-attn==2.3.2+cu117 (from -r temp_requirements.txt (line 65))
  ERROR: HTTP error 404 while getting https://github.com/bdashore3/flash-attention/releases/download/2.3.2-2/flash_attn-2.3.2+cu117-cp310-cp310-win_amd64.whl
ERROR: Could not install requirement flash-attn==2.3.2+cu117 from https://github.com/bdashore3/flash-attention/releases/download/2.3.2-2/flash_attn-2.3.2+cu117-cp310-cp310-win_amd64.whl (from -r temp_requirements.txt (line 65)) because of HTTP error 404 Client Error: Not Found for url: https://github.com/bdashore3/flash-attention/releases/download/2.3.2-2/flash_attn-2.3.2+cu117-cp310-cp310-win_amd64.whl for URL https://github.com/bdashore3/flash-attention/releases/download/2.3.2-2/flash_attn-2.3.2+cu117-cp310-cp310-win_amd64.whl
```

### System Info

```shell
RTX 3090, Win10, System's Python: 3.10.8
```

**Comment by CamiloMM** [comment]
Looking at [the release](https://github.com/bdashore3/flash-attention/releases/tag/2.3.2-2) it seems there exists the following:
`https://github.com/bdashore3/flash-attention/releases/download/2.3.2-2/flash_attn-2.3.2+cu122-cp310-cp310-win_amd64.whl`
but my install is looking for:
`https://github.com/bdashore3/flash-attention/releases/download/2.3.2-2/flash_attn-2.3.2+cu117-cp310-cp310-win_amd64.whl`
So I assume I need to update CUDA or something?

If I run `nvcc --version` I get
```nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Tue_May__3_19:00:59_Pacific_Daylight_Time_2022
Cuda compilation tools, release 11.7, V11.7.64
Build cuda_11.7.r11.7/compiler.31294372_0
```

**Comment by Volty92** [comment]
Same error for me : i reinstalled all with cuda 12.1
Would you like to use CUDA 11.8 instead of 12.1? This is only necessary for older GPUs like Kepler.
If unsure, say "N".

Input (Y/N)> N
CUDA: 12.1
Collecting package metadata (current_repodata.json): done
Solving environment: done

**Comment by CamiloMM** [comment]
Ok now I'm confused. I installed CUDA 12.2-2. `nvcc --version` returns the following, even when inside `cmd_windows.bat`
```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Tue_Aug_15_22:09:35_Pacific_Daylight_Time_2023
Cuda compilation tools, release 12.2, V12.2.140
Build cuda_12.2.r12.2/compiler.33191640_0
```

Then re-ran `update_windows.bat` and it still reports:
```
Ignoring flash-attn: markers 'platform_system == "Windows" and python_version == "3.11"' don't match your environment
Collecting flash-attn==2.3.2+cu117 (from -r temp_requirements.txt (line 65))
  ERROR: HTTP error 404 while getting https://github.com/bdashore3/flash-attention/releases/download/2.3.2-2/flash_attn-2.3.2+cu117-cp310-cp310-win_amd64.whl
ERROR: Could not install requirement flash-attn==2.3.2+cu117 from https://github.com/bdashore3/flash-attention/releases/download/2.3.2-2/flash_attn-2.3.2+cu117-cp310-cp310-win_amd64.whl (from -r temp_requirements.txt (line 65)) because of HTTP error 404 Client Error: Not Found for url: https://github.com/bdashore3/flash-attention/releases/download/2.3.2-2/flash_attn-2.3.2+cu117-cp310-cp310-win_amd64.whl for URL https://github.com/bdashore3/flash-attention/releases/download/2.3.2-2/flash_attn-2.3.2+cu117-cp310-cp310-win_amd64.whl
```

It's still trying to find cu117 even though that's not the current CUDA version.

**Comment by CamiloMM** [comment]
So, it seems the update script specifically downgrades everything to `cu117` or `cu118` when installing. I installed it manually by editing the file and running `python -m pip install -r temp_requirements.txt --upgrade` from `cmd_windows.bat`

This does "fix" the update, but now Exllama2 will not load, with the following:
```
ImportError: DLL load failed while importing flash_attn_2_cuda: The specified module could not be found.
```

**Comment by IJumpAround** [comment]
For a temporary fix lines 64-65 of requirements.txt can be commented out
```
https://github.com/bdashore3/flash-attention/releases/download/2.3.2-2/flash_attn-2.3.2+cu122-cp311-cp311-win_amd64.whl; platform_system == "Windows" and python_version == "3.11"
https://github.com/bdashore3/flash-attention/releases/download/2.3.2-2/flash_attn-2.3.2+cu122-cp310-cp310-win_amd64.whl; platform_system == "Windows" and python_version == "3.10"
```

And things should at least update and load without error if you run the update script again. Doing this will cause ExLlamaV2 to run without flash attention and give a warning, but it will run.
I only quickly tested the exllama + exllama_hf  v1&2 loaders and they all loaded + generated text.

I'm on windows and only removed the windows lines, linux may or may not have the same issue.

**Comment by missionfloyd** [comment]
Delete the `installer_files` folder and run `start_windows.bat`.

**Comment by mykeehu** [comment]
Yes, because the installer has a question asking if you want to use 11.8 or 12.1 cuda. In the upgrade it was left out, "upgrade to 12.1 cuda". After deleting the folder as @missionfloyd suggested, then rebooting with start again, I was able to install the 12.1 cuda and the installation went off. It's just annoying that the 11.8 cudas run to error because of this.

**Comment by CamiloMM** [comment]
> Yes, because the installer has a question asking if you want to use 11.8 or 12.1 cuda. In the upgrade it was left out, "Do you want upgrade to 12.1 cuda?". After deleting the folder as @missionfloyd suggested, then rebooting with start again, I was able to install the 12.1 cuda and the installation went off. It's just annoying that the 11.8 cudas run to error because of this.

So specifically 12.1.1 from the [Nvidia CUDA archive](https://developer.nvidia.com/cuda-toolkit-archive)? It's not something backwards compatible?

**Comment by mykeehu** [comment]
There is no installable file for cuda 11.8 in the path, only for 12.x. On a fresh install, the program asks if you would prefer to use 12.1, but the update didn't offer it, which is why it ran on error. I don't know how to get around this file other than commenting it out if you have 11.8 cuda installed.

**Comment by CamiloMM** [comment]
Ok, I think I got it. In case this problem affects you:
- Delete `installer_files` @missionfloyd mentioned. 
- Install [CUDA 12.2.2](https://developer.nvidia.com/cuda-toolkit-archive), it is the default one on my `$CUDA_PATH` and it works, seems to be backwards compatible with 12.1 so might as well have the more recent version (dunno about 12.3)
- Run `start_windows.bat` and go walk the dog or something.
- `pip3 show flash-attn` from the `cmd_windows.bat` should show that it is installed.
- Exllama2 should work, and there is no message on the console about not having Flash Attention.

**Comment by missionfloyd** [comment]
> * Install [CUDA 12.2.2](https://developer.nvidia.com/cuda-toolkit-archive), it is the default one on my `$CUDA_PATH` and it works, seems to be backwards compatible with 12.1 so might as well have the more recent version (dunno about 12.3)

This step isn't necessary. Or at least, I didn't have to install it.
